### PR TITLE
Bugfix for caching headers

### PR DESCRIPTION
**Pull request by evgeniy-scherbina**

### CODE REVIEW

File: service/middleware.go [hunk 1]
```diff
@@ -250,9 +250,9 @@ func createProxyRequestMiddleware(next http.Handler, config config.Config, servi
 				for headerName, headerValue := range typedCachedResponse.HeaderMap {
 					w.Header().Add(headerName, headerValue)
 				}
-				// add CORS headers
+				// add CORS headers (if not already added)
 				accessControlAllowOriginValue := config.GetAccessControlAllowOriginValue(r.Host)
-				if accessControlAllowOriginValue != "" {
+				if w.Header().Get("Access-Control-Allow-Origin") == "" && accessControlAllowOriginValue != "" {
```
Comment (galxy25) [inline service/middleware.go:255]: add some comments so future devs will take less time to understand this line then it took us to fix please

**Review by galxy25 — APPROVED** [review]

File: service/middleware.go (hunk 1 above)
Comment (galxy25) [reply service/middleware.go:255]: also good to update docs for cache for how it interacts with cors headers

Comment (evgeniy-scherbina) [reply service/middleware.go:255]: yeah, I'll do just want to make sure, it's correct fix

Comment (evgeniy-scherbina) [reply service/middleware.go:255]: @galxy25  https://github.com/Kava-Labs/kava-proxy-service/pull/75

**Pull request merged**